mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -57,6 +57,22 @@ def connection(self) -> taosws.Connection:
             self._connection = self._create_connection()
         return self._connection
 
+    def with_retry_on_closed_connection(self, fn, **kwargs):
+        try:
+            return fn(self.connection, **kwargs)
+        except taosws.QueryError as err:
```

Comment:
I think it won't fix [ML-7806](https://iguazio.atlassian.net/browse/ML-7806), as it doesn't catch `taosws.FetchError`.